src/main/java/io/cdap/e2e/utils/WaitHelper.java

```diff
@@ -44,51 +50,166 @@ public static void waitForPageToLoad(long pageLoadTimeoutInSeconds) {
     SeleniumDriver.getWaitDriver(pageLoadTimeoutInSeconds).until(pageLoadCondition);
   }
 
+  /**
+   * Wait for page to load
+   */
   public static void waitForPageToLoad() {
     logger.info("Waiting for the page to load " +
       "with the default page load timeout: " + ConstantsUtil.PAGE_LOAD_TIMEOUT_SECONDS);
     waitForPageToLoad(ConstantsUtil.PAGE_LOAD_TIMEOUT_SECONDS);
   }
 
+  /**
+   * Wait for element to be present
+   *
+   * @param locator locator of the element
+   * @return WebElement
+   */
   public static WebElement waitForElementToBePresent(By locator) {
     logger.info("Waiting for the element: " + locator + " to be present " +
       "with the Default timeout: " + ConstantsUtil.DEFAULT_TIMEOUT_SECONDS + " seconds");
     return SeleniumDriver.getWaitDriver().until(ExpectedConditions.presenceOfElementLocated(locator));
   }
 
+  /**
+   * Wait for element to be displayed
+   *
+   * @param element WebElement to wait for
+   * @return WebElement
+   */
   public static WebElement waitForElementToBeDisplayed(WebElement element) {
     logger.info("Waiting for the element: " + element + " to be displayed " +
       "with the Default timeout: " + ConstantsUtil.DEFAULT_TIMEOUT_SECONDS + " seconds");
     return SeleniumDriver.getWaitDriver().until(ExpectedConditions.visibilityOf(element));
   }
 
+  /**
+   * Wait for element to be displayed
+   *
+   * @param element          WebElement to wait for
+   * @param timeoutInSeconds timeout
+   * @return WebElement
+   */
   public static WebElement waitForElementToBeDisplayed(WebElement element, long timeoutInSeconds) {
     logger.info("Waiting for the element: " + element + " to be displayed " +
       "with the timeout: " + timeoutInSeconds + " seconds");
     return SeleniumDriver.getWaitDriver(timeoutInSeconds).until(ExpectedConditions.visibilityOf(element));
   }
 
+  /**
+   * Wait for element to be displayed without failing, if the element does not get displayed
+   *
+   * @param locator          Locator of the WebElement
+   * @param timeoutInSeconds timeout
+   */
+  public static boolean waitForElementToBeOptionallyDisplayed(By locator, long timeoutInSeconds) {
+    logger.info("Waiting for the element: " + locator.toString() + " to be optionally displayed " +
+      "with the timeout: " + timeoutInSeconds + " seconds");
+    boolean isElementDisplayed = false;
```

Comment:
I think we don't need this variable, can simply return true if displayed otherwise false

Reply:
Updated.